internal/manifests/compactor.go

```diff
@@ -17,11 +17,10 @@ import (
 )
 
 // BuildCompactor builds the k8s objects required to run Loki Compactor.
-func BuildCompactor(opts Options) []client.Object {
-	return []client.Object{
-		NewCompactorStatefulSet(opts),
-		NewCompactorGRPCService(opts.Name),
-		NewCompactorHTTPService(opts.Name),
+func BuildCompactor(opts Options) (*appsv1.StatefulSet, []client.Object) {
```

Comment:
Let's not break the signature here, please move the configure in here and make this function like `func BuildCompactor(opts Options) ([]client.Object, error)`. It should be an all or nothing builder.

Reply:
Sorry, misunderstood. Should be good now.